### PR TITLE
Mewtwo Adjustment + Resource Restrictions

### PR DESCRIPTION
**Pull request by jaccob101**
Helps bring Mewtwo's recovery, kill power, and combo game down to post-sweep levels. Relies on my float fixes (included). Note that windows were added in regards to float activations to combat super leaf's inaccuracies.
Includes Animation: [mewtwo_assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/11926809/mewtwo_assets.zip)

Mewtwo:
Airdodge:
~[R] Edited animation given by Chrispo (Ult S anim), replaces butt bounce then scoot with a modified special landing anim
https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/d2f882e7-7252-49a8-8240-5b7a6bc314c0

Resource Change:
~[-] Float now depletes jump, and using jump takes away float resource. This matches PM better and reduces his ability to recover from anywhere or combo into the blastzone.
~[-] Teleport in air now depletes float 
- Note that using float makes you not able to act out of teleport.

Float: 
~[+] Duration: 50f -> 60f
- Still shorter than PM
- Theres a 5f window at start of jump to float, and a 10f window at start of float to let go and keep jump. This is to combat wonkiness and inaccuracy from the superleaf state, as in PM tapping up doesn't make you float. Should lessen tap jump users' suffering.

Confusion:
~[/] When interrupting float no longer stalls momentum, similar to confusion after float cancelling in P+.

Dash attack:
~(!) Animation adjusted to move at more consistent rate
~[+] Speed multiplier increased 0.76 -> 0.85

Fair: 
~[-] Hitbox pulled in 1.8 units, still .7 units farther out than vanilla.

Bair:
~[-] KBG: 98/92 -> 94/90
~[-] FAF: 27 -> 30

Down Smash:
~[+] Can now slide during DACDS
~[-] KBG: 122 -> 115

Up Smash: 
~[+] Can now slide during DACUS
~[-] KBG: 118 -> 111

Hitbox Comparison:
Fair: 
Before:
image
After:
image
image

### CODE REVIEW

File: fighters/mewtwo/src/opff.rs [hunk 1]
```diff
@@ -23,16 +23,26 @@ unsafe fn actionable_teleport_air(fighter: &mut L2CFighterCommon, boma: &mut Bat
...
         }
     }
+     //takes away float after 5 frames of jump
+    if boma.get_num_used_jumps() == 2 && (fighter.get_int(*FIGHTER_INSTANCE_WORK_ID_INT_SUPERLEAF_FALL_SLOWLY_FRAME) == VarModule::get_int(boma.object(), vars::common::instance::FLOAT_DURATION)) {
+        if status_kind == *FIGHTER_STATUS_KIND_JUMP_AERIAL && boma.status_frame() > 5 {
+            fighter.set_int(0, *FIGHTER_INSTANCE_WORK_ID_INT_SUPERLEAF_FALL_SLOWLY_FRAME);
+        }
+        else {
+            fighter.set_int(0, *FIGHTER_INSTANCE_WORK_ID_INT_SUPERLEAF_FALL_SLOWLY_FRAME);
+        }
```
Comment (jobrien97) [inline fighters/mewtwo/src/opff.rs]: If/else block not needed here, same logic is being performed either way

File: fighters/common/src/function_hooks/get_param.rs [hunk 2]
```diff
@@ -214,6 +214,15 @@ pub unsafe fn get_param_float_hook(x0 /*boma*/: u64, x1 /*param_type*/: u64, x2
             }
         }
 
+        else if fighter_kind == *FIGHTER_KIND_MEWTWO {
```
Comment (jobrien97) [inline fighters/common/src/function_hooks/get_param.rs]: Would prefer this be done within status script

**Review by jobrien97 — CHANGES_REQUESTED** [review]

**Comment by jobrien97** [comment]
Fair GFX also need to be pulled in more to better match hitbox

**Comment by jaccob101** [comment]
I pulled in the fair effects better, note that it always looks weird super far away or with max zoom even the prev one. Also, idk how to do that in status scripts

File: fighters/common/src/function_hooks/get_param.rs (hunk 2 above)
Comment (jaccob101) [reply fighters/common/src/function_hooks/get_param.rs]: is there another example of someone doing that and is it cleaner than plugging it in there?

**Review by hdr-balance-backroom — DISMISSED** [review]
- Revert up special distance increase
- Remove airdodge lock on up special reappearance
- Revert reappearance cancel frame

File: fighters/common/src/function_hooks/get_param.rs (hunk 2 above)
Comment (jobrien97) [reply fighters/common/src/function_hooks/get_param.rs]: Put this line in an opff func wrapped in a `status_kind == FIGHTER_STATUS_KIND_SPECIAL_S && StatusModule::is_changing && SUPERLEAF_FALL_SLOWLY_FRAME == -1` check:
`WorkModule::off_flag(FIGHTER_MEWTWO_INSTANCE_WORK_ID_FLAG_SPECIAL_S_BUOYANCY)`

Comment (jaccob101) [reply fighters/common/src/function_hooks/get_param.rs]: turning the flag off didnt do anything

**Review by jobrien97 — DISMISSED** [review]
lgtm

**Review by SkewedAskew — APPROVED** [review]

**Pull request merged**